openwisp_controller/pki/api/serializers.py

```diff
@@ -0,0 +1,196 @@
+        fields = [
+            'id',
+            'name',
+            'organization',
+            'notes',
+            'key_length',
+            'digest',
+            'validity_start',
+            'validity_end',
+            'country_code',
+            'state',
+            'city',
+            'organization_name',
+            'organizational_unit_name',
+            'email',
+            'common_name',
+            'extensions',
+            'serial_number',
+            'certificate',
+            'private_key',
+            'passphrase',
+            'created',
+            'modified',
+        ]
+        read_only_fields = ['created', 'modified']
+        extra_kwargs = {
+            'organization': {'required': True},
+            'key_length': {'initial': '2048'},
+            'digest': {'initial': 'sha256'},
+            'validity_start': {'default': default_validity_start()},
+            'validity_end': {'default': default_ca_validity_end()},
+        }
+
+    def validate(self, data):
+        instance = self.instance or self.Meta.model(**data)
+        instance.full_clean()
+        if data.get('certificate') and data.get('private_key'):
+            data = get_import_data(instance)
+        return data
+
+    def validate_validity_start(self, value):
+        if value is None:
+            value = default_validity_start()
+        return value
+
+    def validate_validity_end(self, value):
+        if value is None:
+            value = default_ca_validity_end()
+        return value
+
+
+def CaDetail_fields(fields=None):
+    """
+    Returns the fields for the `CADetailSerializer`.
+    """
+    fields.remove('extensions')
+    fields.remove('passphrase')
+    return fields
+
+
+class CaDetailSerializer(BaseSerializer):
+    class Meta:
+        model = Ca
+        fields = CaDetail_fields(CaListSerializer.Meta.fields[:])
+        read_only_fields = fields[4:]
+
+
+class CaRenewSerializer(CaDetailSerializer):
+    class Meta(CaDetailSerializer.Meta):
+        read_only_fields = CaDetailSerializer.Meta.fields
+
+
+def get_import_data(instance):
+    data = {
+        'name': instance.name,
+        'organization': instance.organization,
+        'key_length': instance.key_length,
+        'digest': instance.digest,
+        'validity_start': instance.validity_start,
+        'validity_end': instance.validity_end,
+        'country_code': instance.country_code,
+        'state': instance.state,
+        'city': instance.city,
+        'organization_name': instance.organization_name,
+        'organizational_unit_name': instance.organizational_unit_name,
+        'email': instance.email,
+        'common_name': instance.common_name,
+        'extensions': instance.extensions,
+        'serial_number': instance.serial_number,
+        'certificate': instance.certificate,
+        'private_key': instance.private_key,
+        'passphrase': instance.passphrase,
+    }
+    return data
+
+
+def CertList_fields(fields=None):
```

Comment:
two doubts here:

- why the default argument is `None`? it looks like there should be no default argument
- why this naming? Capital letter is used for classes, I think this should be named `get_cert_list_fields`

Eg:

```python
def get_cert_list_fields(fields):
    pass
```